### ISSUE TITLE
How to download multiple files using the download extension and then uncompress them under a specific path? 

### ISSUE THREAD

**Issue by lfoppiano** [michel-kraemer/gradle-download-task#394]
I've got the download integrated in an existing task such as: 

```groovy 
task installModels(type: Copy) {
    from "${rootDir}/resources/models/"
    include "**"
    into "${grobidHome}/models/"

    doLast {
        download.run {
            src([
                    'https://grobid.s3.amazonaws.com/dataseer-binary_bert-0.3.1.zip',
                    'https://grobid.s3.amazonaws.com/dataseer-first_bert-0.3.1.zip',
                    'https://grobid.s3.amazonaws.com/dataseer-reuse_bert-0.3.1.zip',
                    'https://grobid.s3.amazonaws.com/datasets-BERT_CRF-0.3.2.zip',
                    'https://grobid.s3.amazonaws.com/context_bert-0.3.2.zip',
                    'https://grobid.s3.amazonaws.com/context_bert_used-0.3.2.zip',
                    'https://grobid.s3.amazonaws.com/context_bert_creation-0.3.2.zip',
                    'https://grobid.s3.amazonaws.com/context_bert_shared-0.3.2.zip'
            ])
            dest "${grobidHome}/models/"
        }
        
    }
}
```

however I would like to decompress all the zip I just copied and remove the zip files. 

I saw the example that uses two tasks: https://github.com/michel-kraemer/gradle-download-task/blob/306172e4c6532e185c8a6a9998bca7d22d2d0c63/examples/groovy/unzip/build.gradle but I cannot really merge the two approaches.

**Comment by lfoppiano** [comment]
I did this: 

```
doLast {
        print "Copy models under grobid-home: ${grobidHome}"
        download.run {
            src([
                    'https://grobid.s3.amazonaws.com/dataseer-binary_bert-0.3.1.zip',
                    'https://grobid.s3.amazonaws.com/dataseer-first_bert-0.3.1.zip',
                    'https://grobid.s3.amazonaws.com/dataseer-reuse_bert-0.3.1.zip',
                    'https://grobid.s3.amazonaws.com/datasets-BERT_CRF-0.3.2.zip',
                    'https://grobid.s3.amazonaws.com/context_bert-0.3.2.zip',
                    'https://grobid.s3.amazonaws.com/context_bert_used-0.3.2.zip',
                    'https://grobid.s3.amazonaws.com/context_bert_creation-0.3.2.zip',
                    'https://grobid.s3.amazonaws.com/context_bert_shared-0.3.2.zip'
            ])
            dest "${grobidHome}/models/"
            overwrite true

            fileTree(dir: "${grobidHome}/models/", includes: ['*.zip']).each { zipFile ->
                from zipTree(zipFile)
            }
            into "${grobidHome}/models/"
        }
    }
```

If you have better solution, feel free ;-)

**Comment by michel-kraemer** [comment]
The most flexible way should be something like this:

```gradle
def models = [
    'https://grobid.s3.amazonaws.com/dataseer-binary_bert-0.3.1.zip',
    'https://grobid.s3.amazonaws.com/dataseer-first_bert-0.3.1.zip',
    'https://grobid.s3.amazonaws.com/dataseer-reuse_bert-0.3.1.zip',
    'https://grobid.s3.amazonaws.com/datasets-BERT_CRF-0.3.2.zip',
    'https://grobid.s3.amazonaws.com/context_bert-0.3.2.zip',
    'https://grobid.s3.amazonaws.com/context_bert_used-0.3.2.zip',
    'https://grobid.s3.amazonaws.com/context_bert_creation-0.3.2.zip',
    'https://grobid.s3.amazonaws.com/context_bert_shared-0.3.2.zip'
]

def installModels = tasks.register("installModels")

models.eachWithIndex { model, index ->
    def downloadTask = tasks.register("downloadModel_$index", Download) {
        src(model)
        dest "${projectDir}/models/"
        onlyIfNewer true
        // overwrite true
    }

    def unzipTask = tasks.register("unzipModel_$index", Copy) {
        dependsOn downloadTask
        from zipTree(downloadTask.get().outputs.files.first())
        into "${projectDir}/models/extracted/"
    }

    installModels.get().dependsOn(unzipTask)
}
```

Creating a separate task for every model has the additional benefit that downloading and unzipping can happen in parallel: as soon as the first model has been downloaded, it can already be unzipped even though the other files are still downloading. Also, using tasks for everything (instead of the `download.run` extension) helps with caching. If you execute the task again and nothing has changed, it will be skipped with an `UP-TO-DATE` message.

Please note that I've replaced `overwrite true` with `onlyIfNewer true` to avoid downloading the models over and over again. This is probably what you want.

Let me know if this helps and if so, please close the issue. Thanks!

**Comment by lfoppiano** [comment]
Thanks!
